ott/geometry/costs.py

```diff
@@ -122,27 +124,111 @@ def __init__(self, dimension: int, **kwargs: Any):
     self._dimension = dimension
     self._sqrtm_kw = kwargs
 
-  def norm(self, x: jnp.ndarray) -> Union[jnp.ndarray, float]:
-    norm = jnp.sum(x[..., 0:self._dimension] ** 2, axis=-1)
-    x_mat = jnp.reshape(
-        x[..., self._dimension:], (-1, self._dimension, self._dimension)
-    )
-
-    norm += jnp.trace(x_mat, axis1=-2, axis2=-1)
+  def norm(self, x):
+    if len(x.shape) == 1:
+      num_components = 1
```

Comment:
i think here you mean batch_size? maybe might be easier to handle this with jnp.atleast_2d to avoid the if